exchanges/okcoin/okcoin_wrapper.go

```diff
@@ -330,37 +332,43 @@ func (o *OKCoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) erro
 
 // UpdateTickers updates the ticker for all currency pairs of a given asset type
 func (o *OKCoin) UpdateTickers(ctx context.Context, a asset.Item) error {
-	if a == asset.Spot {
-		resp, err := o.GetSpotAllTokenPairsInformation(ctx)
-		if err != nil {
-			return err
-		}
-		pairs, err := o.GetEnabledPairs(a)
-		if err != nil {
-			return err
-		}
-		for i := range pairs {
-			for j := range resp {
-				if !pairs[i].Equal(resp[j].InstrumentID) {
-					continue
-				}
-
-				err = ticker.ProcessTicker(&ticker.Price{
-					Last:         resp[j].Last,
-					High:         resp[j].High24h,
-					Low:          resp[j].Low24h,
-					Bid:          resp[j].BestBid,
-					Ask:          resp[j].BestAsk,
-					Volume:       resp[j].BaseVolume24h,
-					QuoteVolume:  resp[j].QuoteVolume24h,
-					Open:         resp[j].Open24h,
-					Pair:         pairs[i],
-					LastUpdated:  resp[j].Timestamp,
-					ExchangeName: o.Name,
-					AssetType:    a})
-				if err != nil {
-					return err
-				}
+	if a != asset.Spot {
+		return fmt.Errorf("%w, asset: %v", asset.ErrNotSupported, a)
+	}
+	tickers, err := o.GetTickers(ctx, "SPOT")
+	if err != nil {
+		return err
+	}
+	enabledPairs, err := o.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		return err
+	}
+	for p := range enabledPairs {
+		for i := range tickers {
+			cp, err := currency.NewPairFromString(tickers[i].InstrumentID)
+			if err != nil {
+				return err
+			}
```

Comment:
Use DeriveFrom method on available pairs and return an error. Then use contains method on the enabled pairs below. This is a bit more performant than calling new pair from string and it catches some malformed pairs.